### PR TITLE
Light client prover rpc

### LINKED ISSUE #1469: Light client prover rpc

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Implements light client prover rpc

## Linked Issues
- Fixes #1469

### CODE REVIEW

File: crates/batch-prover/src/rpc.rs [hunk 1]
```diff
@@ -57,21 +57,21 @@ where
 }
 
 #[rpc(client, server)]
-pub trait ProverRpc {
+pub trait BatchProverRpc {
     /// Generate state transition data for the given L1 block height, and return the data as a borsh serialized hex string.
-    #[method(name = "prover_generateInput")]
+    #[method(name = "batchProver_generateInput")]
```
Comment (jfldde) [inline crates/batch-prover/src/rpc.rs]: Can use `namespace = "batchProver"` in the rpc macro above

File: crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs [hunk 2]
```diff
@@ -206,6 +206,49 @@ pub struct SequencerCommitmentResponse {
...
+    /// The time of the first block in the current epoch (the difficulty adjustment timestamp)
+    pub da_epoch_start_time: u32,
+    /// The UNIX timestamps in seconds of the previous 11 blocks
+    pub da_prev_11_timestamps: [u32; 11],
+    /// Batch proof info from current or previous light client proofs that were not changed and unable to update the state root yet
+    pub unchained_batch_proofs_info: Vec<BatchProofInfo>,
+    /// Last l2 height the light client proof verifies
+    pub last_l2_height: u64,
+    /// Genesis state root of Citrea
+    pub l2_genesis_state_root: [u8; 32],
```
Comment (jfldde) [inline crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs:240]: Shouldn't we use `#[serde(with = "hex::serde")]` there as well ?

File: bin/citrea/provers/risc0/light-client-proof-bitcoin/Cargo.lock
Comment (eyusufatik) [inline bin/citrea/provers/risc0/light-client-proof-bitcoin/Cargo.lock:1]: mistake?

File: bin/citrea/tests/bitcoin_e2e/light_client_test.rs [hunk 3]
```diff
@@ -104,6 +108,31 @@ impl TestCase for LightClientProvingTest {
...
             .unwrap();
 
+        // Expect light client prover to have generated light client proof
+        let lcp = light_client_prover
+            .client
+            .http_client()
+            .get_light_client_proof_by_l1_height(batch_proof_l1_height)
+            .await?;
+        assert!(lcp.is_some());
+        println!("Light client proof generated: {:?}", lcp);
```
Comment (eyusufatik) [inline bin/citrea/tests/bitcoin_e2e/light_client_test.rs]: no print

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**